asab/api/discovery.py

```diff
@@ -62,13 +62,14 @@ async def locate(self, instance_id: str = None, **kwargs) -> list:
 			L.warning("Please provide instance_id, service_id, or other custom id to locate the service(s).")
 			return None
 
-		return [
+		# Each taget can have two records - one for ipv4 and second for ipv6. This information is redundant in the URL format.
+		return set([
 			"http://{}:{}".format(servername, port)
-			for servername, port
+			for servername, port, family
 			in await self._locate(**locate_params)
-		]
+		])
 
-	async def _locate(self, locate_params) -> typing.Set[typing.Tuple]:
+	async def _locate(self, **locate_params) -> typing.Set[typing.Tuple]:
```

Comment:
I intentionally changed this into non-kwargs - there unnecessary overhead linked with this use.

Reply:
but it did not work.